File: marcel-stdlib/src/main/java/marcel/lang/printer/PrintStreamPrinter.java
package marcel.lang.printer;

import lombok.AllArgsConstructor;

import java.io.PrintStream;

@AllArgsConstructor
public class PrintStreamPrinter implements Printer {

  private final PrintStream printStream;

  @Override
  public void print(String s) {
    printStream.print(s);
  }

  @Override
  public void print(Object o) {
    printStream.print(o);
  }

  @Override
  public void println(String s) {
    printStream.println(s);
  }

  @Override
  public void println(Object o) {
    printStream.println(o);
  }

  @Override
  public void println() {
    printStream.println();
  }
}

File: marcel-subprojects/marcel-repl/src/main/kotlin/com/tambapps/marcel/repl/printer/PrintStreamPrinter.kt
package com.tambapps.marcel.repl.printer

import java.io.PrintStream

class PrintStreamPrinter(private val printStream: PrintStream): Printer {

  override suspend fun print(s: CharSequence?) = printStream.print(s)

  override suspend fun println(s: CharSequence?) = printStream.println(s)

  override suspend fun println() = printStream.println()
}